app/assets/javascripts/application.js

```diff
@@ -201,4 +201,34 @@ $(function () {
 
   $("#edit_tab")
     .attr("title", OSM.i18n.t("javascripts.site.edit_disabled_tooltip"));
+
+  document.addEventListener("turbo:frame-load", function (event) {
+    if (event.target.id === "select_language_list") {
+      const $search = $("#language_search");
+
+      if ($search.length) {
+        $search.off("input");
+        $search.on("input", function () {
+          const query = $(this).val().toLowerCase();
+          $(".language-item").each(function () {
+            const text = $(this).text().toLowerCase();
+            $(this).toggle(text.indexOf(query) > -1);
+          });
+        });
+      }
+    }
+  });
+
+  $("#select_language_dialog").on("shown.bs.modal", function () {
+    $("#language_search").val("").trigger("input");
```

Comment:
Would it make sense to grab the focus so that the user can start typing straightaway?

```suggestion
    $("#language_search")
      .focus()
      .val("").trigger("input");
```

(I'm not an accessibility expert, so I might be proposing an a11y sin here).

Reply:
Good idea, I also thought about this before. `.focus()` is deprecated,  I'm using `.trigger("focus")` instead now.